src/wrp-c.c

```diff
@@ -858,7 +859,8 @@ static void decodeRequest( msgpack_object deserialized, int *msgType, char** sou
                         uint32_t cnt = 0;
                         
                         ptr = array.ptr;
-                        *headers_ptr  = ( headers_t *) malloc( sizeof(headers_t));
+                        *headers_ptr  = ( headers_t *) malloc( sizeof(headers_t)
+                                        + sizeof(char *) * array.size);
```

Comment:
The fishy part I fixed in the caller routine, we were allocating the size
of string so the original code in getKey_MsgtypeBin was overwriting an
un-allocated byte (found by valgrind -v.)
My first fix broke it again since now we were writing over the last char in
the buffer ;-(
It is correct now.

As for allocating
                       *headers_ptr = ( headers_t * ) malloc(
sizeof(headers_t)
                                       + sizeof(char *) * 1);
It is still an array of size one, so I believe the memory for it must be
allocated.




On Wed, Jul 20, 2016 at 3:58 PM, Weston Schmidt <notifications@github.com>
wrote:

> In src/wrp-c.c
> <https://github.com/Comcast/wrp-c/pull/16#discussion_r71621897>:
>
> > @@ -858,7 +859,8 @@ static void decodeRequest( msgpack_object deserialized, int *msgType, char** sou
> >                          uint32_t cnt = 0;
> >
> >                          ptr = array.ptr;
> > -                        *headers_ptr  = ( headers_t *) malloc( sizeof(headers_t));
> > +                        *headers_ptr  = ( headers_t *) malloc( sizeof(headers_t)
> > +                                        + sizeof(char *) * array.size);
>
> something seems a bit off - why the need for +1 pointer?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Comcast/wrp-c/pull/16/files/0c92e4198f734dfd42999833c043482cdbe2ecea#r71621897>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AID8OnHM6pgu8YeJNVIHZ43ice3flUU9ks5qXqf7gaJpZM4JRSem>
> .
>